src/GafferArnold/ArnoldShader.cpp

```diff
@@ -69,6 +70,43 @@ ArnoldShader::~ArnoldShader()
 {
 }
 
+Gaffer::Plug *ArnoldShader::correspondingInput( const Gaffer::Plug *output )
+{
+		// better to do a few harmless casts than manage a duplicate implementation
+		return const_cast<Gaffer::Plug *>(
+			const_cast<const ArnoldShader *>( this )->correspondingInput( output )
+		);
+}
+
+const Gaffer::Plug *ArnoldShader::correspondingInput( const Gaffer::Plug *output ) const
+{
+	if( output != outPlug() )
+	{
+		return Shader::correspondingInput( output );
+	}
+
+	ConstCompoundDataPtr metadata = ArnoldShader::metadata();
+	if( !metadata )
+	{
+		return 0;
```

Comment:
Indeed. Matti did that already, so it looks like I've picked up an older version of his work to build on. Investigating...